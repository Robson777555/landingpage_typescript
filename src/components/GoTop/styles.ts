import styled, { css } from 'styled-components';

export const Container = styled.a`
  ${({ theme }) => css`
    position: fixed;
    background: ${theme.colors.primaryColor};
    color: ${theme.colors.white};
    display: flex;
    align-items: center;
    justify-content: center;
    width: 4rem;
    height: 4rem;
    bottom: 2rem;
    right: 2rem;
    border-radius: 50%;
    z-index: 6;
    opacity: 0.7;
    transition: all 0.3s ease;
    cursor: pointer;
    box-shadow: 0 2px 5px rgba(0,0,0,0.2);

    @media (max-width: 768px) {
      width: 3.5rem;
      height: 3.5rem;
      bottom: 1.5rem;
      right: 1.5rem;
    }

    &:hover {
      opacity: 1;
      transform: scale(1.05);
      box-shadow: 0 4px 8px rgba(0,0,0,0.3);
    }

    &:focus {
      outline: 2px solid ${theme.colors.secondaryColor};
      outline-offset: 2px;
    }

    -webkit-tap-highlight-color: transparent;
    touch-action: manipulation;
  `}
`;