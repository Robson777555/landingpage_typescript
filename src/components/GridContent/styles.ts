// src/components/GridContent/styles.js
import styled, { css } from 'styled-components';

export const Container = styled.div`
  ${({ theme }) => css`
    font-family: ${theme.fonts.family.default};
    text-align: center;
    max-width: 58rem;
    margin: 0 auto;
    padding: 0 ${theme.spacings.large};
  `}
`;

export const HtmlContainer = styled.div`
  ${({ theme }) => css`
    font-family: ${theme.fonts.family.default};
    margin: ${theme.spacings.xhuge} 0;
    text-align: left;

    & > * {
      margin-bottom: ${theme.spacings.medium};
    }

    h1, h2, h3, h4, h5, h6 {
      margin: ${theme.spacings.large} 0;
    }

    pre {
      display: inline-block;
      margin: 0 auto;
      background: rgba(0,0,0,0.1);
      padding: ${theme.spacings.small};
      border-radius: ${theme.spacings.xsmall};
      overflow-x: auto;
      max-width: 100%;
    }

    blockquote {
      border-left: 3px solid ${theme.colors.primaryColor};
      padding-left: ${theme.spacings.medium};
      font-style: italic;
      color: ${theme.colors.mediumGray};
    }

    a {
      color: ${theme.colors.primaryColor};
      text-decoration: none;
      &:hover {
        text-decoration: underline;
      }
    }
  `}
`;