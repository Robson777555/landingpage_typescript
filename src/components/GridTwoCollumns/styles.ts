// src/components/GridTwoCollumns/styles.js
import styled, { css } from 'styled-components';

export const Container = styled.div`
  ${({ theme }) => css`
    display: flex;
    align-items: center;
    justify-content: space-between;
    max-width: 1200px;
    margin: 0 auto;
    padding: ${theme.spacings.large};
    gap: ${theme.spacings.large};

    @media ${theme.media.lteMedium} {
      flex-direction: column;
      text-align: center;
      justify-content: center;
      gap: ${theme.spacings.large};
    }
  `}
`;

export const TextContainer = styled.div`
  ${({ theme }) => css`
    display: flex;
    flex-direction: column;
    gap: ${theme.spacings.medium};

    /* Estilos para mobile */
    @media ${theme.media.lteMedium} {
      align-items: center;
    }
  `}
`;

export const ImageContainer = styled.div`
  ${({ theme }) => css`
    display: flex;
    justify-content: center;
    align-items: center;
    padding: ${theme.spacings.medium};
    max-width: 400px;
    width: 100%;

    img, .next-image {
      max-width: 100%;
      height: auto;
      width: 100%;
      object-fit: contain;
    }

    @media ${theme.media.lteMedium} {
      padding: ${theme.spacings.large} 0;
      max-width: 300px;
    }
  `}
`;

export const Title = styled.h2`
  font-size: 2.4rem;
  font-weight: bold;
  text-align: center;
  word-break: break-word;
  white-space: normal;
  width: 100%;
  line-height: 1.2;
  margin-bottom: ${({ theme }) => theme.spacings.medium};

  @media (max-width: 768px) {
    font-size: 2rem;
  }
`;
