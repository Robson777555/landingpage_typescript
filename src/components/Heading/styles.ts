import styled, { css, DefaultTheme } from 'styled-components';

interface TitleProps {
  $colorDark?: boolean;
  $size?: 'small' | 'medium' | 'big' | 'huge';
  $uppercase?: boolean;
}

const mediaFontSize = (theme: DefaultTheme) => css`
  @media (max-width: 768px) {
    font-size: ${theme.fonts.sizes.xlarge};
  }
`;

export const Title = styled('h2').attrs<TitleProps>(({ as = 'h2' }) => ({
  as,
}))<TitleProps>`
  font-weight: bold;
  text-align: center;
  white-space: normal;
  word-break: break-word;
  width: 100%;
  line-height: 1.2;
  margin-bottom: ${({ theme }) => theme.spacings.medium};
  color: ${({ theme, $colorDark }) => 
    $colorDark ? theme.colors.primaryColor : theme.colors.white};
  text-transform: ${({ $uppercase }) => $uppercase ? 'uppercase' : 'none'};
  font-size: ${({ theme, $size }) => {
    switch($size) {
      case 'small': return theme.fonts.sizes.small;
      case 'medium': return theme.fonts.sizes.medium;
      case 'big': return theme.fonts.sizes.large;
      case 'huge': return theme.fonts.sizes.xlarge;
      default: return theme.fonts.sizes.xlarge;
    }
  }};

  @media (max-width: 768px) {
    font-size: 2rem;
  }

  ${({ theme }) => mediaFontSize(theme)}
`;