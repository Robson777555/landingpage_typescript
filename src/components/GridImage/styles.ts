// src/components/GridImage/styles.js
import styled, { css } from 'styled-components';
import { Container as TextComponent } from '../TextComponent/styles';

export const Container = styled.div`
  ${({ theme }) => css`
    font-family: ${theme.fonts.family.default};
    max-width: 1200px;
    margin: 0 auto;
    padding: ${theme.spacings.xlarge};

    > ${TextComponent} {
      margin-bottom: ${theme.spacings.xhuge};
    }
  `}
`;

export const Grid = styled.div`
  ${({ theme }) => css`
    display: grid;
    grid-template-columns: repeat(auto-fill, minmax(280px, 1fr));
    gap: ${theme.spacings.large};
    margin-top: ${theme.spacings.large};

    @media (max-width: 768px) {
      grid-template-columns: 1fr;
      gap: ${theme.spacings.large};
      justify-items: center;
    }
  `}
`;

export const GridElement = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;
  ${({ theme }) => css`
    border-radius: ${theme.spacings.small};
    overflow: hidden;
    box-shadow: 0 4px 8px rgba(0,0,0,0.1);
    transition: transform 300ms ease-in-out;
    width: 100%;
    max-width: 340px;
    margin: 0 auto;
    padding: ${theme.spacings.medium};

    &:hover {
      transform: translateY(-5px);
    }

    img, .next-image {
      max-width: 220px;
      width: 100%;
      height: auto;
      margin: 0 auto;
      display: block;
    }

    h2, h3, h4, h5, h6 {
      font-size: 2.4rem;
      font-weight: bold;
      text-align: center;
      white-space: normal;
      word-break: break-word;
      width: 100%;
      line-height: 1.2;
      margin-bottom: ${theme.spacings.small};
    }

    @media (max-width: 768px) {
      padding: ${theme.spacings.small};
      max-width: 98vw;
      min-width: 0;
      h2, h3, h4, h5, h6 {
        font-size: 2rem;
        font-weight: bold;
        text-align: center;
        white-space: normal;
        word-break: break-word;
        width: 100%;
        line-height: 1.2;
      }
    }
  `}
`;
