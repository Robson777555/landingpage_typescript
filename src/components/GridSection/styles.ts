import styled, { css } from 'styled-components';
import { Title as HeadingContainer } from '../Heading/styles';
import { Container as TextComponent } from '../TextComponent/styles';

export const Container = styled.div`
  ${({ theme }) => css`
    font-family: ${theme.fonts.family.default};
    max-width: 1200px;
    margin: 0 auto;
    padding: ${theme.spacings.xlarge};

    ${TextComponent} {
      margin-bottom: ${theme.spacings.xhuge};
      text-align: center;
    }
  `}
`;

export const Grid = styled.div`
  ${({ theme }) => css`
    counter-reset: grid-counter;
    display: grid;
    grid-template-columns: repeat(auto-fill, minmax(280px, 1fr));
    gap: ${theme.spacings.large};
    margin-top: ${theme.spacings.large};

    @media (max-width: 768px) {
      grid-template-columns: 1fr;
      gap: ${theme.spacings.xlarge};
    }
  `}
`;

export const GridElement = styled.div`
  ${({ theme }) => css`
    position: relative;
    padding: ${theme.spacings.medium};
    background: ${theme.colors.white};
    border-radius: ${theme.spacings.small};
    box-shadow: 0 2px 4px rgba(0, 0, 0, 0.1);
    transition: transform 0.3s ease;
    overflow: hidden;

    &:hover {
      transform: translateY(-5px);
      box-shadow: 0 4px 8px rgba(0, 0, 0, 0.15);
    }

    ${HeadingContainer} {
      position: relative;
      left: 0;
      margin-bottom: ${theme.spacings.medium};
      font-size: 2.4rem;
      text-align: center;
      word-break: break-word;
      white-space: normal;
      font-weight: bold;
      @media (max-width: 768px) {
        font-size: 2rem;
      }
    }

    ${HeadingContainer}::before {
      counter-increment: grid-counter;
      content: counter(grid-counter);
      position: absolute;
      font-size: 4rem;
      top: -2rem;
      left: 0.5rem;
      transform: rotate(0deg);
      color: ${theme.colors.mediumGray};
      opacity: 0.3;
      z-index: 0;
      @media (max-width: 768px) {
        font-size: 2.5rem;
        top: -1.2rem;
        left: 0.5rem;
      }
    }

    ${TextComponent} {
      text-align: left;
      font-size: 1.2rem;
      @media (max-width: 768px) {
        font-size: 1rem;
      }
    }
  `}
`;

export const Title = styled.h3`
  font-size: 2.4rem;
  font-weight: bold;
  text-align: center;
  word-break: break-word;
  white-space: normal;
  width: 100%;
  line-height: 1.2;
  margin-bottom: ${({ theme }) => theme.spacings.medium};

  @media (max-width: 768px) {
    font-size: 2rem;
  }
`;
