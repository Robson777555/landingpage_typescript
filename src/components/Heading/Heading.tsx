import React from 'react';
import { Title } from './styles';

type HeadingSize = 'small' | 'medium' | 'big' | 'huge';
type HeadingAs = 'h1' | 'h2' | 'h3' | 'h4' | 'h5' | 'h6';

interface HeadingProps {
  children: React.ReactNode;
  colorDark?: boolean;
  as?: HeadingAs;
  size?: HeadingSize;
  uppercase?: boolean;
}

export const Heading = ({
  children,
  colorDark = true,
  as = 'h1',
  size = 'huge',
  uppercase = false,
}: HeadingProps) => {
  return (
    <Title
      as={as}
      $colorDark={colorDark}
      $size={size}
      $uppercase={uppercase}
      data-testid="heading"
    >
      {children}
    </Title>
  );
};