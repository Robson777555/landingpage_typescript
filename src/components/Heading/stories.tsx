import { Meta, StoryObj } from "@storybook/react";
import { Heading } from "./Heading";
import { theme } from "../../styles/theme";

const meta: Meta<typeof Heading> = {
  title: "Components/Heading",
  component: Heading,
  tags: ["autodocs"],
  argTypes: {
    children: {
      control: "text",
      defaultValue: "Edite este texto",
    },
    colorDark: {
      control: "boolean",
      defaultValue: true,
    },
    as: {
      control: "select",
      options: ["h1", "h2", "h3", "h4", "h5", "h6"],
      defaultValue: "h1",
    },
    size: {
      control: "select",
      options: ["small", "medium", "big", "huge"],
      defaultValue: "huge",
    },
    uppercase: {
      control: "boolean",
      defaultValue: false,
    },
  },
  args: {
    children: "Edite este texto",
    colorDark: true,
    as: "h1",
    size: "huge",
    uppercase: false,
  },
};

export default meta;

type Story = StoryObj<typeof Heading>;

export const Default: Story = {};

export const Light: Story = {
  args: {
    colorDark: true,
    children: "Texto escuro",
  },
  parameters: {
    backgrounds: {
      default: "light",
      values: [{ name: "light", value: theme.colors.white }],
    },
  },
};

export const Dark: Story = {
  args: {
    colorDark: false,
    children: "Texto claro",
  },
  parameters: {
    backgrounds: {
      default: "dark",
      values: [{ name: "dark", value: theme.colors.primaryColor }],
    },
  },
};

export const Uppercase: Story = {
  args: {
    uppercase: true,
    children: "TEXTO EM MAIÚSCULAS",
  },
};

export const CustomSize: Story = {
  args: {
    size: "medium",
    children: "Tamanho médio",
  },
};
