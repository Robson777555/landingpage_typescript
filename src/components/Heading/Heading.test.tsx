import React from 'react';
import { render, screen } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';
import { theme } from '../../styles/theme';
import { Heading } from './Heading';
import { describe, it, expect } from 'vitest';

describe('Heading Component', () => {
  const renderHeading = (props = {}) => {
    const defaultProps = {
      children: 'Test Heading',
    };

    return render(
      <ThemeProvider theme={theme}>
        <Heading {...defaultProps} {...props} />
      </ThemeProvider>
    );
  };

  it('should render with default values', () => {
    renderHeading();

    const heading = screen.getByTestId('heading');
    expect(heading).toBeInTheDocument();
    expect(heading.tagName.toLowerCase()).toBe('h1');
    expect(heading).toHaveTextContent('Test Heading');
    expect(heading).toHaveStyle(`color: ${theme.colors.primaryColor}`); // Cor base do tema
    expect(heading).toHaveStyle('text-transform: none');
  });

  it('should render as different HTML tags', () => {
    const headings = ['h1', 'h2', 'h3', 'h4', 'h5', 'h6'] as const;
    headings.forEach((as) => {
      // Limpa o DOM antes de cada render
      const { unmount } = renderHeading({ as });
      const heading = screen.getByTestId('heading');
      expect(heading.tagName.toLowerCase()).toBe(as);
      unmount();
    });
  });

  it('should render children content correctly', () => {
    const customText = 'Custom Heading Text';
    renderHeading({ children: customText });
    expect(screen.getByTestId('heading')).toHaveTextContent(customText);
  });

  it('should apply uppercase style when uppercase prop is true', () => {
    renderHeading({ uppercase: true });
    const heading = screen.getByTestId('heading');
    expect(heading).toHaveStyle('text-transform: uppercase');
  });

  it('should apply dark color when colorDark is false', () => {
    renderHeading({ colorDark: false });
    const heading = screen.getByTestId('heading');
    expect(heading).toHaveStyle(`color: ${theme.colors.white}`); // Cor do texto claro
  });

  it('should apply different sizes correctly', () => {
    const sizes = ['small', 'medium', 'big', 'huge'] as const;
    sizes.forEach((size) => {
      // Limpa o DOM antes de cada render
      const { unmount } = renderHeading({ size });
      const heading = screen.getByTestId('heading');
      let expectedFontSize = '';
      switch (size) {
        case 'small': expectedFontSize = theme.fonts.sizes.small; break;
        case 'medium': expectedFontSize = theme.fonts.sizes.medium; break;
        case 'big': expectedFontSize = theme.fonts.sizes.large; break;
        case 'huge': expectedFontSize = theme.fonts.sizes.xlarge; break;
        default: expectedFontSize = theme.fonts.sizes.xlarge;
      }
      expect(heading).toHaveStyle(`font-size: ${expectedFontSize}`);
      unmount();
    });
  });
  it ('Snapshot of page', () => {
    renderHeading();
    const heading = screen.getByTestId('heading');
    expect(heading).toMatchSnapshot();
  })
});