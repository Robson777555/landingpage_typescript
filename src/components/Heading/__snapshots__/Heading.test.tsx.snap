// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Heading Component > Snapshot of page 1`] = `
<h1
  class="sc-bXdtCk ksXdHw"
  data-testid="heading"
>
  Test Heading
</h1>
`;
